src/abstractdataframe/abstractdataframe.jl

```diff
@@ -2060,10 +2060,30 @@ function Missings.disallowmissing(df::AbstractDataFrame,
     for i in axes(df, 2)
         x = df[!, i]
         if i in idxcols
-            if !error && Missing <: eltype(x) && any(ismissing, x)
-                y = x
+            local y
+            if Missing <: eltype(x)
+                if error
+                    try
+                        y = disallowmissing(x)
+                    catch e
+                        row = findfirst(ismissing, x)
+                        if row !== nothing
+                            col = _names(df)[i]
+                            throw(ArgumentError("Missing value found in column " *
+                                                ":$col in row $row"))
+                        else
+                            rethrow(e)
+                        end
+                    end
+                else
+                    if any(ismissing, x)
+                        y = x
+                    else
+                        y = disallowmissing(x)
+                    end
+                end
```

Comment:
Instead of this, we could just do `error || continue` in the `if row !== nothing` branch above.

Reply:
We could not do `continue` because we still need to execute line 2088, so instead we would need to write inside `if row !== nothing`:

```
if error
    col = _names(df)[i]
    throw(ArgumentError("Missing value found in column :$col in row $row"))
else
    y = x
end
```

Do you think it would be simpler to understand by the potential readers of the code?
I used this explicit logic because when I read the old condition:
```
if !error && Missing <: eltype(x) && any(ismissing, x)
```
I had to spend some time to make sure it was OK (although I think I have written it some time ago).

Therefore I would vote for leaving things as they are as it is I think easier to understand if some user wants to inspect the code.

Reply:
OK. But I'd use `try... catch` when `!error` too.

Reply:
I have changed `disallowmissing` and `disallowmissing!` the way that I think you wanted.

Reply:
OK. But I'd use `try... catch` when `!error` too.